### ISSUE TITLE
[Suggestion] Ability to disable 'patching smdh'

### ISSUE THREAD

**Issue by Slattz** [imthe666st/NTRSharp#6]
Hi! I'll try keep it short:

Basically, while using ACNL (w/ update) w/ this debugger, loading a new area will softlock the game.
It's apparently because '_**the client patches the SMDH to see all active processes, which usually corrupts the FS process**_'. I'm bringing this issue to light now since you're active on this project. It's been an issue since the NTRClient project (Which I'm very grateful for! It's made cheat searching w/o GW so much easier. I can't wait to see how this new project turns out!).

So, what I'm proposing is this: An option to disable the process name check via patching smdh, but instead just show the pids in the processes dropdown. This will apparently stop the game from softlocking. 

I know this project is new but I think it would be a great feature to have, even if it's only for ACNL. Thanks for reading this! 🙂

**Comment by imthe666st** [comment]
As far as I know NTR patches the smdh by itself.  
I'm currently downloading the patch for ACNL to see if I can recreate the issue on my end.  

Does the issue also occur if the debugger is not connected and you're playing just with NTR in the background?

**Comment by Slattz** [comment]
nope, the game doesn't softlock when ntr is enabled but the debugger isn't connected. It also doesn't happen when connected to the original debugger either. It seems to be something with your debugger(s) & I was told it was patching smdh that caused it.

**Comment by imthe666st** [comment]
Alright, did some quick testing.  
It appears the game only crashes once the processes have been listed. I'll now test if it crashes because of the memlayout, or the process list. (or both, who knows).

**Comment by imthe666st** [comment]
On the default NTRClient:  

1. Start NTR ( I use 3.2 )  
2. Enable the debugger
3. Connect to your 3DS
4. Start the game
5. Enter "memlayout(*PID_HERE*)"

This will result in a crash.

**Comment by imthe666st** [comment]
Simply listing the processes doesn't seem to trigger the behavior.  
I could add an option to disable the Memregion support. This would also disable the Memregion check.

**Comment by Slattz** [comment]
Doing what you described crashed my game. Huh, I was told it was because of how every process had its smdh patched compared to the original NTRClient. What would disabling the Memregion support do? But if it fixes the problem then maybe it's worth a shot.

**Comment by Slattz** [comment]
Well if it will fix the problem then it's worth doing imo. 

Would it be possible to set custom memregions, instead of NTR retrieving them you can just add them in client side using a txt file or something? That way you could still use the dropdown, dump all memory, etc but it wouldn't need to communicate with the 3ds itself to get them which would stop the softlocks. 
Its just a theory, I don't know how NTR handles all that stuff so my theory mightn't be accurate.

**Comment by imthe666st** [comment]
Changed some stuff in the code today, I'll try my best to implement it tomorrow.

**Comment by imthe666st** [comment]
Uploaded a new release, which can be found [here](https://github.com/imthe666st/NTRSharp/releases).  

If your issue has been resolved feel free to close this issue.

**Comment by Slattz** [comment]
I don't think you uploaded the release lol
Anyway, I compiled from source, couldn't get it to run, turns out I needed a restrictions.txt in the same path as the exe. I don't know if this is a bug or not.

In what way was the issue meant to be resolved? I still get the softlock when loading a new area and the memregion dropdown box isn't disabled so I don't really know if it's fixed or not tbh.

Though I have found a workaround that mostly works, I can start the client and get the memregions for acnl, then disconnect. Then I restart my 3ds and connect again to the client but now the memlayout command doesn't need to be sent since the client already has them. Now I can load new areas since the memlayout command wasn't sent, which is what is causing the problem.
